Monika After Story/game/script-topics.rpy

```diff
@@ -3221,6 +3224,55 @@ label monika_lovecounter_aff:
 
 default persistent._mas_last_monika_ily = None
 
+label monika_ilym_fight_start:
+    $ ilym_times_till_win = renpy.random.randint(6,10)
+    $ ilym_count = 0
+    $ ilym_quip = "I love you more, " + player + "!"
+    m 3hubfb "[ilym_quip]"
+    jump monika_ilym_fight
+
+label monika_ilym_fight:
+    $ _history_list.pop()
+    menu:
+        m "[ilym_quip]{fast}"
+        "No, I love you more!":
+            if ilym_count < ilym_times_till_win:
+                if renpy.random.randint(1,3) == 1:
```

Comment:
Not sure you need this random chance bit here at all. Seems to me you may as well just have everything grouped as quips.

(Also optimizes this flow and makes it easier to expand in future)